src/pick.ts

```diff
@@ -0,0 +1,66 @@
+import IterableInfer from "./types/IterableInfer";
+import { isAsyncIterable, isIterable } from "./_internal/utils";
+import toArray from "./toArray";
+
+function sync<T extends object, U extends Iterable<keyof T>>(
+  iterable: U,
+  obj: T,
+): Pick<T, IterableInfer<U>> {
+  const keys = Array.isArray(iterable) ? iterable : Array.from(iterable);
+  return Object.fromEntries(
+    Object.entries(obj).filter(([k]) => keys.includes(k as keyof T)),
+  ) as Pick<T, IterableInfer<U>>;
+}
+
+async function async<T extends object, U extends AsyncIterable<keyof T>>(
+  iterable: U,
+  obj: T,
+): Promise<Pick<T, IterableInfer<U>>> {
+  const keys = await toArray(iterable);
+  return Object.fromEntries(
+    Object.entries(obj).filter(([k]) => keys.includes(k as keyof T)),
+  ) as any;
```

Comment:
How about making the keys object?
```ts
{
   [keys]:true
   ...
}
```

so we can use this instead of `keys.includes`
`Object.entries(obj).filter(([k]) => keys[k]))`

Reply:
Good suggestion. I will change `keys` to the instance of `Set`. `Set.prototype.has` method also has `O(1)` time complexity.